### PR TITLE
List by topics v1.0b

### PR DESCRIPTION
**Pull request by heejet**

### CODE REVIEW

File: src/main/java/seedu/clialgo/CLIAlgo.java [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
+    }
 
+    /** Continuously reads in the user input until the exit command is executed */
+    private void run() {
+        while (true) {
+            String fullCommand = ui.getUserInput();
+            Command command = parser.parse(fullCommand, topicManager);
+            command.execute(topicManager, ui, fileManager);
+        }
     }
```
Comment (ong-ck) [inline src/main/java/seedu/clialgo/CLIAlgo.java:28]: Great use of SLAP!

File: src/main/java/seedu/clialgo/TopicManager.java [hunk 2]
```diff
@@ -34,4 +41,60 @@ public TopicManager() {
...
+
+    /** Checks if a specified topic has no notes stored in it. */
+    public boolean isTopicEmpty(String topic) {
+        return topics.get(topic).isEmpty();
+    }
+
+    /** Checks if a given note name have been used before. */
+    public boolean isRepeatedNote(String noteName) {
+        return this.allNotes.contains(noteName);
+    }
```
Comment (ong-ck) [inline src/main/java/seedu/clialgo/TopicManager.java:58]: Perhaps can do JUnit testing for these functions as well

**Review by ong-ck — COMMENTED** [review]
Looks good to merge!

**Comment by nikkiDEEE** [comment]
LGTM

**Pull request merged**